dali/pipeline/operators/decoder/nvjpeg_decoder_cpu.h

```diff
@@ -55,9 +55,26 @@ class nvJPEGDecoderCPUStage : public Operator<CPUBackend> {
                                                     GetFormat(output_image_type_)));
       NVJPEG_CALL(nvjpegDecodeParamsSetAllowCMYK(decode_params_[i], true));
     }
+
+
+    // We allocate the pinned buffers here to alliviate the first run
+    const int nbuffers = spec.GetArgument<int>("cpu_prefetch_queue_depth") * batch_size_;
+    buffer_initial_pool_.resize(nbuffers);
+    for (auto& state : buffer_initial_pool_) {
+      state = std::make_shared<StateNvJPEG>(handle_, decoder_host_, decoder_hybrid_);
+    }
+
+    const auto output_format = GetFormat(output_image_type_);
+    for (auto& state : buffer_initial_pool_) {
+      state->nvjpeg_backend = NVJPEG_BACKEND_HYBRID;
+      WarmUpNvJPEG(handle_, *state, decoder_host_, output_format);
+      state->nvjpeg_backend = NVJPEG_BACKEND_GPU_HYBRID;
+      WarmUpNvJPEG(handle_, *state, decoder_hybrid_, output_format);
+    }
   }
 
   virtual ~nvJPEGDecoderCPUStage() noexcept(false) {
+    buffer_initial_pool_.clear();
```

Comment:
Could you remove the `noexcept(false)`, `NVJPEG_CALL`s and manage the error codes differently? [Boyscout rule](https://deviq.com/boy-scout-rule/) ;)